ts/src/bitget.ts

```diff
@@ -2634,44 +2649,71 @@ export default class bitget extends Exchange {
+                if (!isMarketOrder) {
+                    throw new ExchangeError (this.id + ' createOrder() bitget stopLoss or takeProfit orders must be market orders');
+                }
+                request['holdSide'] = (side === 'buy') ? 'long' : 'short';
+            }
+            const reduceOnly = this.safeValue (params, 'reduceOnly', false);
+            if (isTriggerOrder) {
                 request['triggerPrice'] = this.priceToPrecision (symbol, triggerPrice);
                 if (price !== undefined) {
                     request['executePrice'] = this.priceToPrecision (symbol, price);
                 }
-                method = 'privateMixPostPlanPlacePlan';
-            }
-            if (isStopLossOrTakeProfit) {
-                if (!isMarketOrder) {
-                    throw new ExchangeError (this.id + ' createOrder() bitget stopLoss or takeProfit orders must be market orders');
+                if (side === 'buy') {
+                    request['side'] = 'open_long';
+                } else if (side === 'sell') {
+                    request['side'] = 'open_short';
+                } else {
+                    request['side'] = side;
                 }
-                if (isStopLossOrder) {
-                    request['triggerPrice'] = this.priceToPrecision (symbol, stopLossPrice);
+                method = 'privateMixPostPlanPlacePlan';
+            } else if (isStopLossOrTakeProfitTrigger) {
+                if (isStopLossTriggerOrder) {
+                    request['triggerPrice'] = this.priceToPrecision (symbol, stopLossTriggerPrice);
                     request['planType'] = 'loss_plan';
-                } else if (isTakeProfitOrder) {
-                    request['triggerPrice'] = this.priceToPrecision (symbol, takeProfitPrice);
+                } else if (isTakeProfitTriggerOrder) {
+                    request['triggerPrice'] = this.priceToPrecision (symbol, takeProfitTriggerPrice);
                     request['planType'] = 'profit_plan';
                 }
-                request['holdSide'] = (side === 'buy') ? 'long' : 'short';
                 method = 'privateMixPostPlanPlaceTPSL';
+            } else if (isStopLossOrTakeProfit) {
+                if (isStopLoss) {
+                    request['triggerPrice'] = this.priceToPrecision (symbol, stopLoss);
+                    request['planType'] = 'pos_loss';
+                } else if (isTakeProfit) {
+                    request['triggerPrice'] = this.priceToPrecision (symbol, takeProfit);
+                    request['planType'] = 'pos_profit';
+                }
+                method = 'privateMixPostPlanPlacePositionsTPSL';
             } else {
                 if (reduceOnly) {
                     request['side'] = (side === 'buy') ? 'close_short' : 'close_long';
                 } else {
-                    request['side'] = (side === 'buy') ? 'open_long' : 'open_short';
+                    if (side === 'buy') {
```

Comment:
@Dan-krm let's avoid this logic `request['side'] = (side === 'buy') ? 'close_short' : 'close_long';` whenever possible because let's say the user provides `close_short`, by doing that, it would default to `open_short`  we should translate buy/sell to the appropriate value but we should keep any other value as is